.github/workflows/cli-ci.yml

```diff
@@ -23,12 +23,12 @@ jobs:
         with:
           go-version: "^1.20.0"
       - uses: actions/checkout@v3
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3
-        with:
-          version: v1.51
-          args: -v --timeout 5m0s
-          working-directory: cli/azd
+      # - name: golangci-lint
+      #   uses: golangci/golangci-lint-action@v3
+      #   with:
+      #     version: v1.51
+      #     args: -v --timeout 5m0s
+      #     working-directory: cli/azd
```

Comment:
The `AsyncContext` can digest more than string messages.  The `Progress` type `P` can be of cany struct that you need.  Today we are only using it for progress messages that contain a string/timestamp but you could imagine we could introduce additional context data such as message type, verbosity, etc that we could use to drive the type of data we show in the UX.

Reply:
Can we make `Progress` to be the `console` ?   
Then, the async task can directly use the console instead of requiring the extra type?

Reply:
We should avoid tightly coupling our UX to our backend service components so that both can evolve separately.  The UX console printing should happen in a single place for a command without having to search through various components to change different parts of the output.